main.go

```diff
@@ -0,0 +1,84 @@
+package main
+
+import (
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials/stscreds"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/ministryofjustice/analytical-platform-auth0-user-exports/auth"
+	"log"
+	"os"
+	"time"
+
+	session2 "github.com/aws/aws-sdk-go/aws/session"
+	env "github.com/ministryofjustice/analytical-platform-auth0-user-exports/default_env_value"
+	connection "github.com/ministryofjustice/analytical-platform-auth0-user-exports/get_connection"
+	"github.com/ministryofjustice/analytical-platform-auth0-user-exports/job"
+)
+
+var (
+	apiUrl         = env.SetDefaultEnvValue("API_URL", "https://alpha-analytics-moj.eu.auth0.com")
+	clientId       = os.Getenv("CLIENT_ID")
+	clientSecret   = os.Getenv("CLIENT_SECRET")
+	connectionName = env.SetDefaultEnvValue("CONNECTION_NAME", "github")
+	platform       = os.Getenv("ENV")
+	session        = session2.Must(session2.NewSession())
+	creds          = stscreds.NewCredentials(session, env.SetDefaultEnvValue("ROLE_ARN", "arn:aws:iam::593291632749:role/auth0-user-data-write"))
+	s3svc          = s3.New(session, &aws.Config{Credentials: creds, Region: aws.String("eu-west-1"), CredentialsChainVerboseErrors: aws.Bool(true)})
+)
+
+func main() {
+
+	tkn, err := auth.GetToken(apiUrl, clientId, clientSecret)
+	if err != nil {
+		log.Fatalf("Error while retrieving access token: %s", err)
+	}
+
+	connectionId, err := connection.GetConnection(fmt.Sprintf("%s/api/v2/connections", apiUrl), tkn.AccessToken, connectionName)
+	if err != nil {
+		log.Fatalf("Error while retrieving connection id: %s", err)
+	}
+
+	jobConfig := fmt.Sprintf(`{
+		"connection_id": "%s",
+		"format": "csv", 
+		"fields": [
+			{"name": "email"}, { "name": "nickname", "export_as": "username"}, {"name": "created_at", "export_as": "joined"}
+		]
+	}`, *connectionId)
+
+	bulkUserExportJob, err := job.CreateJob(fmt.Sprintf("%s/api/v2/jobs/users-exports", apiUrl), tkn.AccessToken, jobConfig)
+	if err != nil {
+		log.Fatalf("Error occurred while creating job: \n%s", err)
+	}
+
+	resultLocation, err := job.WaitForJobCompletion(fmt.Sprintf("%s/api/v2/jobs", apiUrl), tkn.AccessToken, *bulkUserExportJob)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	exportData, err := job.GetUserExport(*resultLocation)
+	if err != nil {
+		log.Fatalf("Error trying to download userdata: \n%s", err)
+	}
+
+	bucket := env.SetDefaultEnvValue("BUCKET", "auth0-userdata")
+	key := fmt.Sprintf("userdata-%s.csv", time.Now().Format("02-01-2006"))
+
+	if platform == "aws" {
+
+		result, err := job.UploadUserExportToS3(s3svc, exportData, bucket, key)
+		if err != nil {
+			log.Print(fmt.Errorf("Error while uploading userdata to S3: \n%s", err))
+		} else {
+			log.Printf("FILE=%s successfully uploaded to BUCKET=%s, VERSIONID=%p", key, bucket, result.VersionId)
+		}
```

Comment:
usually you don't use the else if it's the happy path as it makes the code harder to scan.

In general (oversimplifying)good things happen at the first level of indentation, bad things happen at the second level of indentation, e.g.

```golang
res, err := myFunc()
if err != nil {
  // error handling
}

// more good things happen here
```